docs/sources/tempo/configuration/network/tls.md

```diff
@@ -96,3 +96,127 @@ The above structure can be set on the following receiver configurations:
+
+### Configure TLS with Helm
+
+To configure TLS with the Helm chart, you must have a TLS key-pair and CA certificate stored in a Kubernetes secret. The following example mounts a secret called `tempo-distributed-tls` into the pods at `/tls` and modifies the configuration of Tempo to make use of the files. In this example the Tempo components share a single TLS certificate. Note that the `tls_server_name` configuration must match the certificate.
+
+```yaml
+compactor:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+distributor:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+ingester:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+metricsGenerator:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+querier:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+queryFrontend:
+  extraVolumeMounts:
+    - mountPath: /tls
+      name: tempo-distributed-tls
+  extraVolumes:
+    - name: tempo-distributed-tls
+      secret:
+        secretName: tempo-distributed-tls
+tempo:
+  readinessProbe:
+    httpGet:
+      scheme: HTTPS
+  structuredConfig:
+    memberlist:
+      tls_ca_path: /tls/ca.crt
+      tls_cert_path: /tls/tls.crt
+      tls_enabled: true
+      tls_key_path: /tls/tls.key
+      tls_server_name: t-dist.tempo-distributed.svc.cluster.znet
+    distributor:
+      receivers:
+        otlp:
+          protocols:
+            grpc:
+              tls:
+                ca_file: /tls/ca.crt
+                cert_file: /tls/tls.crt
+                key_file: /tls/tls.key
+    ingester_client:
+      grpc_client_config:
+        tls_ca_path: /tls/ca.crt
+        tls_cert_path: /tls/tls.crt
+        tls_enabled: true
+        tls_key_path: /tls/tls.key
+        tls_server_name: tempo-distributed.trace.svc.cluster.local
+    metrics_generator_client:
+      grpc_client_config:
+        tls_ca_path: /tls/ca.crt
+        tls_cert_path: /tls/tls.crt
+        tls_enabled: true
+        tls_key_path: /tls/tls.key
+        tls_server_name: tempo-distributed.trace.svc.cluster.local
+    querier:
+      frontend_worker:
+        grpc_client_config:
+          tls_ca_path: /tls/ca.crt
+          tls_cert_path: /tls/tls.crt
+          tls_enabled: true
+          tls_key_path: /tls/tls.key
+          tls_server_name: tempo-distributed.trace.svc.cluster.local
```

Comment:
This client is the only one that supports two ways of setting TLS. Through this method, but also through [here](https://github.com/grafana/helm-charts/blob/main/charts/tempo-distributed/values.yaml#L758). If you see value in unification, I would remove the latter since at least for me it just caused confusion when setting up TLS (given, back then I didn't know of the `structuredConfig` option)

Reply:
That sounds reasonable to me.  Would you like to PR for this?  If not I can follow up.  We'll want to mark it a breaking change, and validate that the structuredConfig is the right approach in all cases.  Wouldn't mind at least other eyes on that portion.